mlir/test/Dialect/Mesh/simplifications.mlir

```diff
@@ -0,0 +1,131 @@
+// RUN: mlir-opt -test-mesh-simplifications %s | FileCheck %s
```

Comment:
What happened if the `all_reduce` has multi users?

Reply:
Yes, it's a complex problem. A common approach is to leave it unsimplified. XLA has [a pass](https://github.com/openxla/xla/blob/main/xla/service/all_reduce_reassociate.h) with similar logic, maybe we can check its logic there

Reply:
I made it not do the simplification in such a scenario.